src/ray/status.h

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+//
+// A Status encapsulates the result of an operation.  It may indicate success,
+// or it may indicate an error with an associated error message.
+//
+// Multiple threads can invoke const methods on a Status without
+// external synchronization, but if any of the threads may call a
+// non-const method, all threads accessing the same Status must use
+// external synchronization.
+
+// Adapted from Apache Arrow, Apache Kudu, TensorFlow
+
+#ifndef RAY_STATUS_H_
+#define RAY_STATUS_H_
+
+#include <cstring>
+#include <iosfwd>
+#include <string>
+
+#include "ray/util/macros.h"
+#include "ray/util/visibility.h"
+
+// Return the given status if it is not OK.
+#define RAY_RETURN_NOT_OK(s)           \
```

Comment:
Let's be consistent with a bunch of codebase out there and use `RAY_RETURN_IF_ERROR`?

Reply:
I'd prefer this one since it it the one used by arrow and we already use it in a bunch of places in plasma (which was merged with arrow). But no big deal, if you strongly prefer the other way we can also rename it :)